File: CS/Form1.cs


using System.Windows.Forms;
using System.Data;
using System;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.LookAndFeel;
using DevExpress.Utils;
using DevExpress.Utils.Drawing;
using System.Drawing;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid;
namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        private DefaultPaintHelper _DefaultPaintHelper;
             
        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}{1}{0}{1}{0}", i, Environment.NewLine), i, 3 - i, DateTime.Now.AddDays(i) });
            return tbl;
        }


        public Form1()
        {
            InitializeComponent();
            gridView1.CustomDrawCell += gridView1_CustomDrawCell;
            gridControl1.DataSource = CreateTable(20);
            gridView1.RowHeight = 40;
            gridView1.Columns[0].ColumnEdit = new RepositoryItemMemoEdit();
            _DefaultPaintHelper = new DefaultPaintHelper(gridControl1);
        }

        void gridView1_CustomDrawCell(object sender, RowCellCustomDrawEventArgs e)
        {
            if (!(e.Column.ColumnEdit is RepositoryItemMemoEdit))
                return;
            e.Handled = true;
            _DefaultPaintHelper.DefaultDrawContent(e);
            MyScrollBarPaintHelper.DrawScrollBar(e.Cache, e.Bounds);
        }

  
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports System.Data
Imports System
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.LookAndFeel
Imports DevExpress.Utils
Imports DevExpress.Utils.Drawing
Imports System.Drawing
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid
Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Private _DefaultPaintHelper As DefaultPaintHelper

		Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(DateTime))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}{1}{0}{1}{0}", i, Environment.NewLine), i, 3 - i, DateTime.Now.AddDays(i) })
			Next i
			Return tbl
		End Function


		Public Sub New()
			InitializeComponent()
			AddHandler gridView1.CustomDrawCell, AddressOf gridView1_CustomDrawCell
			gridControl1.DataSource = CreateTable(20)
			gridView1.RowHeight = 40
			gridView1.Columns(0).ColumnEdit = New RepositoryItemMemoEdit()
			_DefaultPaintHelper = New DefaultPaintHelper(gridControl1)
		End Sub

		Private Sub gridView1_CustomDrawCell(ByVal sender As Object, ByVal e As RowCellCustomDrawEventArgs)
			If Not(TypeOf e.Column.ColumnEdit Is RepositoryItemMemoEdit) Then
				Return
			End If
			e.Handled = True
			_DefaultPaintHelper.DefaultDrawContent(e)
			MyScrollBarPaintHelper.DrawScrollBar(e.Cache, e.Bounds)
		End Sub


	End Class
End Namespace